docs/2.build/4.web3-apps/chain-ui.md

```diff
@@ -0,0 +1,77 @@
+
+:::
+
+### Pros
+
+- Decentralized front-ends are less susceptible to single points of failure, making them more resistant to attacks and server downtimes
+- By decentralizing the hosting of your front-end, you minimize the risk of censoring or blocking content by centralized authorities.
+- Users might trust a decentralized application more, knowing that it operates on a transparent and immutable blockchain.
+- Data displayed on the front-end is more likely to be accurate and tamper-proof since it's typically fetched directly from the blockchain.
+
+### Cons
+
+- Implementing a decentralized front-end can be more complex than traditional web development, requiring knowledge of additional technologies like [IPFS](https://ipfs.tech), [Arweave](https://arweave.org), or blockchain interactions.
+- Decentralized networks can face issues such as latency or lower speeds compared to traditional centralized servers, potentially affecting user experience.
+The ecosystem for developing decentralized frontends is still maturing, which means there might be fewer tools and libraries available than for traditional web development.
+- While decentralized storage costs have been decreasing, they can still be higher than traditional hosting, especially if the dApp generates a lot of data transactions.
+
+## How it works
+
+A standard workflow of the chain deployed solution would be:
+- Building, packing, and compressing the web app frontend
+- Uploading the web app bundle files using the `FileStore` smart contract
+- Serving the web app using a CDN gateway
+
+Once the web app has been uploaded to the blockchain, a web [Gateway](#gateway) converts browser requests to blockchain [RPC](#rpc) calls, so users can access your decentralized application.
+
+### Gateway
+
+A gateway server is required to convert browser resource requests to RPC calls. For convenience, gateways may be deployed to cloud hosts, but this causes some centralization. It can be mitigated by having multiple gateways run by different parties. It is important that a user trust the gateway provider however, since the gateway is ultimately what is in control of the experience being served to the user's browser.
+
+### RPC
+
+An [RPC provider](../../5.api/rpc/providers.md) services requests for chain data. For resilience, gateways should ideally be capable of falling back to a different provider in the event the primary provider is experiencing degraded service.
+
+## Considerations
+
+Compared to classic Web2 hosting solutions, the cost structure of a chain-deployed dApp is very different.
+
+When deploying on chain, you pay for the storage space of your compressed bundles. This means that it's critical to pay attention to the size of dependencies you include in your application, as packages that are bloated or do not support proper tree-shaking will directly affect your cost.
```

Comment:
> pay for the storage space of your compressed bundles

This would be a good place to call out that this a one time cost 😎

Reply:
Good call. Changed to:

```
When deploying on chain, you pay a one-time fee for the storage space of your compressed bundles.
```